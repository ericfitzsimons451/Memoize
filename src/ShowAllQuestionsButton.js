import React from 'react';
import './ShowAllQuestionsButton.scss';

function ShowAllQuestionsButton(props) {
  return(
    <div>
      <button type="button" className="show-all-questions-button" onClick={props.showAllQuestions}>Reset All Questions</button>
    </div>  
  )
}

export default ShowAllQuestionsButton;
