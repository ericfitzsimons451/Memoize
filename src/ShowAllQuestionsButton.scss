.show-all-questions-button {
  height: 60px;
  width: 200px;
  border-radius: 30px;
  font-size: 1.5rem;
  margin: 15px;
  &:hover {
    cursor: pointer;
  }
}