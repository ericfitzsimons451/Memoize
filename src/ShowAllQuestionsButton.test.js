import React from 'react';
import ShowAllQuestionsButton from './ShowAllQuestionsButton.js';
import { shallow } from 'enzyme';

const mockShowAllQuestions = jest.fn();

describe('ShowAllQuestionsButton', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ShowAllQuestionsButton showAllQuestions={mockShowAllQuestions} />
      )
  })

  it('should match the snapshot with all data passed down', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should register a click when RESET ALL QUESTIONS is clicked', () => {
    wrapper.find('.show-all-questions-button').simulate('click')
    expect(mockShowAllQuestions).toBeCalled()
  })
})