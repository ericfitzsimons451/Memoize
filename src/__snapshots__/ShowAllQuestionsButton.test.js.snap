// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowAllQuestionsButton should match the snapshot with all data passed down 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShowAllQuestionsButton
    showAllQuestions={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="show-all-questions-button"
        onClick={[MockFunction]}
        type="button"
      >
        Reset All Questions
      </button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Reset All Questions",
        "className": "show-all-questions-button",
        "onClick": [MockFunction],
        "type": "button",
      },
      "ref": null,
      "rendered": "Reset All Questions",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="show-all-questions-button"
          onClick={[MockFunction]}
          type="button"
        >
          Reset All Questions
        </button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset All Questions",
          "className": "show-all-questions-button",
          "onClick": [MockFunction],
          "type": "button",
        },
        "ref": null,
        "rendered": "Reset All Questions",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
